toplevel/byte/topmain.ml

```diff
@@ -156,19 +156,25 @@ let prepare ppf =
       Format.fprintf ppf "Uncaught exception: %s\n" (Printexc.to_string x);
       false
 
-(* If [name] is "", then the "file" is stdin treated as a script file. *)
 let file_argument name =
+  let filename =
+    let open Toploop in
+    match name with
+    | File name -> name
+    | String _ | Stdin -> ""
```

Comment:
This logic is duplicated many times, maybe it could go to `Toploop.filename_of_input : input -> string`?